### PR TITLE
Switch to cross-compilation for `linux_aarch64` and `linux_ppc64le`

### PR DESCRIPTION
**Pull request by bdice**
This PR switches to cross-compilation for `linux_aarch64` and `linux_ppc64le`.

See https://github.com/conda-forge/cuda-feedstock/issues/25

### CODE REVIEW

File: recipe/run_nvcc_tests.bat [hunk 1]
```diff
@@ -0,0 +1,10 @@
+@echo on
+
+nvcc --version
+if errorlevel 1 exit 1
+
+nvcc --verbose test.cu
+if errorlevel 1 exit 1
+
+cmake -S . -B ./build -G=Ninja && cmake --build ./build -v
+if errorlevel 1 exit 1
```
Comment (jakirkham) [inline recipe/run_nvcc_tests.bat:10]: Also this covers all the same Windows tests as before in this script

**Review by jakirkham — COMMENTED** [review]
Noted a few changes below

All notable changes are to how testing is done

In particular when cross-compiling, we are not able to run `nvcc`. So this adds logic to check if cross-compilation is being done and skip tests accordingly

That said, the tests can be run on the packages produced when run on the appropriate platform using [`conda build --test <pkg file>`]( https://docs.conda.io/projects/conda-build/en/latest/user-guide/tutorials/building-conda-packages.html#automated-testing )

Also native builds still have tests run as before

**Review by adibbley — APPROVED** [review]

File: recipe/run_nvcc_tests.sh [hunk 2]
```diff
@@ -1,8 +1,28 @@
 #!/bin/bash
 set -e
 set -x
+
+test_platform="linux-$(uname -p)"
+if [[ "${test_platform}" != "${target_platform}" ]]
+then
+  echo "Skipping NVCC testing as current platform is ${test_platform}, whereas; NVCC is built for ${target_platform}."
+  exit 0
+fi
```
[the remark this answers is not in the document]
Comment (isuruf) [reply recipe/run_nvcc_tests.sh:10]: QEMU emulation does not work?

[the remark this answers is not in the document]
Comment (jakirkham) [reply recipe/run_nvcc_tests.sh]: No `nvcc --version` segfaults:

```
+ nvcc --version
/home/conda/feedstock_root/build_artifacts/cuda-nvcc-impl-split_1712783024858/test_tmp/run_test.sh: line 8:  2784 Segmentation fault      (core dumped) nvcc --version
```

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=913312&view=logs&j=6494c7a6-a019-59a3-6f3d-9bab8ca96717&t=18c91bca-a25a-5f89-34d4-93bbec4021bc&l=1496

**Pull request merged**